nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/RunMongoAggregation.java

```diff
@@ -194,6 +199,8 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
 
             if (batch.size() > 0) {
                 writeBatch(buildBatch(batch), flowFile, context, session, attrs, REL_RESULTS);
+            } else {
```

Comment:
Would be cleaner to do `if (batch.isEmpty()) {} else {}`

Reply:
I've reviewed my logic and I found a flaw. Something could be processed in the while loop and in the end the batch is empty. In this case I can't redirect to EMPTY. I've fixed this including a control flag "doneSomething".